### PR TITLE
chore: migrate from kuttl to chainsaw (linodemachine-controller)

### PR DESCRIPTION
**Pull request by eddycharly**
Migrate from kuttl to chainsaw.

### CODE REVIEW

**Comment by eddycharly** [comment]
cc @eljohnson92

**Comment by eddycharly** [comment]
Folks, can anyone give a look ?

File: e2e/linodemachine-controller/vpc-integration/chainsaw-test.yaml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    - script:
+        env:
+        - name: TARGET_API
+          value: api.linode.com
+        - name: TARGET_API_VERSION
+          value: v4beta
+        - name: URI
+          value: linode/instances
+        - name: FILTER
+          value: '{"tags":"vpc-machine-test"}'
```
Comment (eljohnson92) [inline e2e/linodemachine-controller/vpc-integration/chainsaw-test.yaml:36]:
```suggestion
          value: '{"tags":"machine-test-vpc"}'
```

File: e2e/linodemachine-controller/vpc-integration/chainsaw-test.yaml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+          set -e
+          curl -s \
+            -H "Authorization: Bearer $LINODE_TOKEN" \
+            -H "X-Filter: $FILTER" \
+            -H "Content-Type: application/json" \
+            "https://$TARGET_API/$TARGET_API_VERSION/$URI"
+        check:
+          ($error): ~
+          (json_parse($stdout)):
+            active: true
```
Comment (eljohnson92) [inline e2e/linodemachine-controller/vpc-integration/chainsaw-test.yaml:68]: This was the only failure in the tests, I might need your help crafting the correct syntax here. The returned JSON for reference is
```
{
  "data": [
    {
      "id": 423512,
      "label": "machine-test-vpc",
      "description": "",
      "region": "us-sea",
      "subnets": [
        {
          "id": 8765312,
          "label": "test-subnet",
          "ipv4": "10.0.0.0/24",
          "ipv6": null,
          "linodes": [
            {
              "id": 123452346,
              "interfaces": [
                {
                  "id": 542346,
                  "active": true
                }
              ]
            }
          ],
          "created": "2024-03-07T14:33:12",
          "updated": "2024-03-07T14:33:12"
        }
      ],
      "created": "2024-03-07T14:33:12",
      "updated": "2024-03-07T14:33:12"
    }
  ],
  "page": 1,
  "pages": 1,
  "results": 1
}

```
I think we want something like this
`data.linodes[0].interfaces[0].active: true` but I'm not sure the exact syntax to get that result

Comment (eddycharly) [reply e2e/linodemachine-controller/vpc-integration/chainsaw-test.yaml:68]: this should do it
```yaml
data:
- subnets:
  - linodes:
    - interface:
      - active: true
```

Comment (eddycharly) [reply e2e/linodemachine-controller/vpc-integration/chainsaw-test.yaml]: Fixed

**Comment by eddycharly** [comment]
@eljohnson92 i addressed your comments, can you give it a try ?

File: e2e/linodemachine-controller/vpc-integration/chainsaw-test.yaml [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    - script:
+        env:
+        - name: TARGET_API
+          value: api.linode.com
+        - name: TARGET_API_VERSION
+          value: v4beta
+        - name: URI
+          value: linode/instances
+        - name: FILTER
+          value: '{"tags":"machine-test-vpc"}'
```
Comment (eljohnson92) [inline e2e/linodemachine-controller/vpc-integration/chainsaw-test.yaml:36]:
```suggestion
          value: '{"tags":"vpc-machine-test"}'
```

File: e2e/linodemachine-controller/vpc-integration/chainsaw-test.yaml [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    - script:
+        env:
+        - name: TARGET_API
+          value: api.linode.com
+        - name: TARGET_API_VERSION
+          value: v4beta
+        - name: URI
+          value: linode/instances
+        - name: FILTER
+          value: '{"tags":"linodecluster-test"}'
```
Comment (eljohnson92) [inline e2e/linodemachine-controller/vpc-integration/chainsaw-test.yaml:98]:
```suggestion
          value: '{"tags":"machine-test-vpc"}'
```

File: e2e/linodemachine-controller/vpc-integration/chainsaw-test.yaml [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    - script:
+        env:
+        - name: TARGET_API
+          value: api.linode.com
+        - name: TARGET_API_VERSION
+          value: v4beta
+        - name: URI
+          value: vpcs
+        - name: FILTER
+          value: '{"label":"linodevpc-test"}'
```
Comment (eljohnson92) [inline e2e/linodemachine-controller/vpc-integration/chainsaw-test.yaml:119]:
```suggestion
          value: '{"label":"machine-test-vpc"}'
```

**Comment by eddycharly** [comment]
@eljohnson92 done

**Comment by eljohnson92** [comment]
LGTM, working locally. good to merge this in

**Review by eljohnson92 — APPROVED** [review]

**Comment by eddycharly** [comment]
> LGTM, working locally. good to merge this in

Cool, thanks !
I will open another one tomorrow.

**Pull request merged**